### PR TITLE
Refactor firmata

### PR DESCRIPTION
**Pull request by zankich**
This pull request refactors the firmata gobot adaptor, as well as separates the firmata protocol implementation into a separate package.

### CODE REVIEW

File: platforms/firmata/client/client.go [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+	AnalogMappingQuery       byte = 0x69
+	AnalogMappingResponse    byte = 0x6A
+	StringData               byte = 0x71
+	I2CRequest               byte = 0x76
+	I2CReply                 byte = 0x77
+	I2CConfig                byte = 0x78
+	FirmwareQuery            byte = 0x79
+	I2CModeWrite             byte = 0x00
+	I2CModeRead              byte = 0x01
+	I2CmodeContinuousRead    byte = 0x02
```
Comment (deadprogram) [inline platforms/firmata/client/client.go]: Case?

**Comment by deadprogram** [comment]
A much needed refactor! Good stuff.

**Comment by zankich** [comment]
@deadprogram fixed the typo

**Comment by deadprogram** [comment]
:zero: :one:

**Pull request merged**